api/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemData.java

```diff
@@ -114,6 +115,87 @@ public interface CustomItemData {
      */
     @NonNull Set<String> tags();
 
+    /**
+     * Gets the stack size of the item.
+     *
+     * Returns 0 if not set. When not set (or 0), it defaults to the stack count of the Java item when based of a vanilla item, or 64 when registering a non-vanilla item.
+     *
+     * Note that, to copy Java behaviour, setting the stack size of an item to a value above 1 will set the max damage to 0. If a max damage value above 0 was explicitly set, an exception will be thrown.
+     *
+     * @return the stack size of the item
+     */
+    @NonNegative
+    int stackSize();
```

Comment:
Can be applied to a handful of other methods: Use "based on" or "based off". Also add paragraph tags to actually render the space when it is desired.

Specific to `stackSize()` and `maxDamage()`: Please make it clear that it is not these methods that fallback to a default, it is Geyser's registration system that uses the described fallback behaviour. I would replace the usage of "it"

Reply:
I adjusted the documentation, thanks!